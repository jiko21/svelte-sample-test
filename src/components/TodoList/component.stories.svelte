<script>
  import { Meta, Template, Story } from '@storybook/addon-svelte-csf';
  import TodoList from './component.svelte';
</script>

<Meta
  title="components/TodoList"
  component={TodoList}
  argTypes={{
    lists: { control: 'text' },
    onRemove: { action: 'onRemove' },
  }}
/>

<!-- More on component templates: https://storybook.js.org/docs/svelte/writing-stories/introduction#using-args -->
<Template let:args>
  <TodoList {...args} on:remove={args.onClick} />
</Template>

<!-- More on args: https://storybook.js.org/docs/svelte/writing-stories/args -->
<Story
  name="Default"
  args={{
    lists: [
      {
        id: 1,
        content: 'test1test1',
      },
      {
        id: 2,
        content: 'test2test2',
      },
      {
        id: 3,
        content: 'test3test3test3test3test3test3test3test3test3test3test3test3',
      },
    ],
  }}
/>
