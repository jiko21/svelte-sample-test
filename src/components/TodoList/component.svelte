<script lang="ts">
  import { createEventDispatcher } from 'svelte';
  import Button from '../Button';
  import Card from '../Card';
  export let lists = [];
  const dispatch = createEventDispatcher();
  const removeItem = (id: number) => {
    dispatch('remove', {
      id: id,
    });
  };
</script>

<ul>
  {#each lists as item (item.id)}
    <li>
      <Card>
        <div class="card-container">
          <span>{item.content}</span>
          <Button
            on:click={() => removeItem(item.id)}
            label={'remove'}
            testId={`remove-${item.id}`}
            style="danger"
          />
        </div>
      </Card>
    </li>
  {/each}
</ul>

<style>
  ul {
    list-style: none;
    padding: 0;
  }

  li {
      margin: 4px 0;
  }

  .card-container {
      align-items: center;
      display: flex;
      justify-content: space-between;
  }

  span {
      color: #3c3c3c;
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
  }
</style>
