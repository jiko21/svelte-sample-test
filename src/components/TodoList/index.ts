import TodoList from './component.svelte';

export default TodoList;
