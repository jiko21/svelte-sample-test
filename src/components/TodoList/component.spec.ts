import { fireEvent, screen } from '@testing-library/dom';
import { render } from '@testing-library/svelte';

import TodoList from './component.svelte';

const dispatchMock = jest.fn();

jest.mock('svelte', () => {
  const origin = jest.requireActual('svelte');

  return {
    ...origin,
    createEventDispatcher: () => dispatchMock,
  };
});

const LISTS = [
  {
    id: 0,
    content: 'test1test1',
  },
  {
    id: 1,
    content: 'test2test2',
  },
  {
    id: 2,
    content: 'test3test3',
  },
];

describe('components/TodoList', () => {
  beforeEach(() => jest.clearAllMocks());

  it('correctly render', () => {
    const component = render(TodoList, {
      lists: LISTS,
    });
    expect(component.container).toMatchSnapshot();
  });

  it('call function correctly', async () => {
    const removeMock = jest.fn();
    const { component, container } = render(TodoList, {
      lists: LISTS,
    });
    component.$on('remove', removeMock);

    await fireEvent.click(screen.getByTestId('remove-1'));

    expect(dispatchMock).toBeCalledWith('remove', {
      id: 1,
    });
  });
});
