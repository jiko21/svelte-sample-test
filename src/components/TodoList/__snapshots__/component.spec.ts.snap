// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/TodoList correctly render 1`] = `
<body>
  <div>
    <ul
      class="svelte-c9lhan"
    >
      <li
        class="svelte-c9lhan"
      >
        <div
          class="card svelte-ab6tor"
        >
          <div
            class="card-container svelte-c9lhan"
          >
            <span
              class="svelte-c9lhan"
            >
              test1test1
            </span>
             
            <button
              class="danger svelte-1ohb8zt"
              data-testid="remove-0"
            >
              remove
            </button>
          </div>
        </div>
         
      </li>
      <li
        class="svelte-c9lhan"
      >
        <div
          class="card svelte-ab6tor"
        >
          <div
            class="card-container svelte-c9lhan"
          >
            <span
              class="svelte-c9lhan"
            >
              test2test2
            </span>
             
            <button
              class="danger svelte-1ohb8zt"
              data-testid="remove-1"
            >
              remove
            </button>
          </div>
        </div>
         
      </li>
      <li
        class="svelte-c9lhan"
      >
        <div
          class="card svelte-ab6tor"
        >
          <div
            class="card-container svelte-c9lhan"
          >
            <span
              class="svelte-c9lhan"
            >
              test3test3
            </span>
             
            <button
              class="danger svelte-1ohb8zt"
              data-testid="remove-2"
            >
              remove
            </button>
          </div>
        </div>
         
      </li>
    </ul>
  </div>
</body>
`;
